docs/tutorials/posts/gpt-oss-python.md

```diff
@@ -0,0 +1,255 @@
+    pass@1[avg-of-16] | 30          | 17091      | 1983        | 94.79%           | 0.00%
+    majority@16       | 30          | 17091      | 1983        | 96.67%           | 0.00%
+    pass@16           | 30          | 17091      | 1983        | 100.00%          | 0.00%
+
+    ----------------------------------------- aime25 ----------------------------------------
+    evaluation_mode   | num_entries | avg_tokens | gen_seconds | symbolic_correct | no_answer
+    pass@1[avg-of-16] | 30          | 21070      | 2330        | 94.17%           | 0.00%
+    majority@16       | 30          | 21070      | 2330        | 100.00%          | 0.00%
+    pass@16           | 30          | 21070      | 2330        | 100.00%          | 0.00%
+    ```
+
+!!! tip
+    you can also run `ns summarize_results --cluster <cluster> <output_dir>` to re-print the summary of all metrics.
+
+## Synthetic data generation
+
+Switching from evaluation to SDG is really simple! Here is an example of how you can re-generate solutions for
+[OpenMathReasoning](../../releases/openmathreasoning/index.md) dataset using gpt-oss-120b with Python enabled.
+You can adjust the commands accordingly to switch to a different reasoning regime or disable Python.
+
+Let's first download a set of problems from OpenMathReasoning dataset as a jsonl file.
+
+```python
+from datasets import concatenate_datasets, load_dataset
+
+def remove_proofs(example):
+    return example['problem_type'] != 'converted_proof'
+
+dataset = load_dataset("nvidia/OpenMathReasoning")
+
+dataset['cot'] = dataset['cot'].remove_columns(
+    ['generation_model', 'generated_solution', 'inference_mode', 'used_in_kaggle']
+)
+dataset['additional_problems'] = dataset['additional_problems'].remove_columns(
+    ['generation_model', 'generated_solution', 'inference_mode', 'used_in_kaggle']
+)
+full_data = concatenate_datasets([dataset['cot'], dataset['additional_problems']])
+full_data = full_data.filter(remove_proofs, num_proc=20)
+
+full_data.to_json("math-problems.jsonl")
+```
+
+Then run generation command using this file as an input (you might need to upload it on Slurm if you prepared data locally).
+
+```python
+from nemo_skills.pipeline.cli import generate, wrap_arguments
+
+cluster = "slurm"  # change this to match your cluster config name
+
+generate(
+    ctx=wrap_arguments(
+        # we are using fewer tokens than max context length as code output isn't accounted for
+        "++inference.tokens_to_generate=120000 "
+        # recommended inference settings including prompt config
+        "++inference.temperature=1.0 "
+        "++inference.top_p=1.0 "
+        "++prompt_config=gpt-oss/math "
+        # we currently implement native Python code tool through text completions API
+        # as we found alternative implementations to have issues.
+        # We will switch to the official responses API when those are addressed
```

Comment:
did you mean "when the support is added"?

Reply:
I think they technically support it in some way, it's just buggy. At least it was when I last tried

Reply:
Sorry I meant the language, the phrasing is a bit awkward right now.